hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azure/TestSyncableDataOutputStream.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.fs.azure;
+
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.junit.Test;
+
+public class TestSyncableDataOutputStream {
+
+  @Test
+  public void testCloseWhenFlushThrowingIOException() throws Exception {
+    MockOutputStream out = new MockOutputStream();
+    SyncableDataOutputStream sdos = new SyncableDataOutputStream(out);
+    out.flushThrowIOE = true;
+    try {
```

Comment:
use LambdaTestUtils.intercept

Reply:
Done.  Also added try with resource statement also in the test using LambdaTestUtils.